oc_cdtapi/ForemanAPI.py

```diff
@@ -35,6 +35,22 @@ def __init__(self):
         deploy_on = 1
         self.apiversion = int(os.getenv("FOREMAN_API_VERSION", "1") or "1")
         self.defs = class_defaults(exp_date, location_id, hostgroup, deploy_on)
+        self.foreman_version = None
+
+    def get_foreman_version(self):
+        """
+        lazy init foreman_version
+        """
+        logging.debug('Reached get_foreman_version')
+        if self.foreman_version is None:
+            response = self.get("status").json()
+            self.foreman_version = response.get("version")
+            logging.debug("version: [%s]".format(self.foreman_version))
+        return self.foreman_version
+
+    def get_foreman_version_major(self):
+        version = self.get_foreman_version()
+        return int(version.split('.')[0])
```

Comment:
There is `packaging` stable tool for this. Please re-use it instead of self-writing tool.
```
from packaging import version
...
_major = version.parse("1.0.0").major
```

Do not forget to check `packaging` dependency is specified in `install_requires` key in the `setup.py`